challenges/intermediate/crack-caesar-cipher/solutions/aleattene/solution.js

```diff
@@ -0,0 +1,87 @@
+/*
+JS solution for challenge: "Crack Caesar Cipher"
+To test the solution, type from CLI: node solution.js (required node.js)
+*/
+
+
+const {decode} = require("./encode_decode.js");
+fs = require('fs');
+
+// Name of the files to crack
+files = ["Shakespeare-Hamlet.txt", "Shakespeare-Macbeth.txt", "Shakespeare-Romeo-And-Juliet.txt"];
+
+// Theoretical frequency of characters present in a text written in English
+let theoreticalFrequency = {
+    'a': 8.167, 'b': 1.492, 'c': 2.782, 'd': 4.253, 'e': 12.702, 'f': 2.228, 'g': 2.015, 'h': 6.094,
+    'i': 6.966, 'j': 0.153, 'k': 0.772, 'l': 4.025, 'm': 2.406,  'n': 6.749, 'o': 7.507, 'p': 1.929,
+    'q': 0.095, 'r': 5.987, 's': 6.327, 't': 9.056, 'u': 2.758,  'v': 0.978, 'w': 2.360, 'x': 0.150,
+    'y': 1.974, 'z': 0.074
+};
+
+let i = 0;
+while (i < files.length) {
+
+    // Data Structures for the occurrence of characters and their real frequency in the analyzed text
+    let occurrences = {}, realFrequency = {};
+
+    // Current file to be cracked
+    let file =  files[i];
+
+    // File reading
+    fs.readFile('../../cases/' + file, 'utf8',
```

Comment:
This is good!
I'd just suggest you wrap the entire cracking function as a standalone function taking the text input as an argument, so it can run both on client and server.

Reply:
Done.